server/src/main/java/io/unitycatalog/server/service/credential/aws/AwsCredentialVendor.java

```diff
@@ -0,0 +1,81 @@
+package io.unitycatalog.server.service.credential.aws;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.persist.utils.ServerPropertiesUtils;
+import io.unitycatalog.server.service.credential.CredentialContext;
+import java.time.Duration;
+import java.util.Map;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.model.Credentials;
+
+public class AwsCredentialVendor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AwsCredentialVendor.class);
+
+  private final Map<String, S3StorageConfig> s3Configurations;
+
+  public AwsCredentialVendor() {
+    this.s3Configurations = ServerPropertiesUtils.getInstance().getS3Configurations();
+  }
+
+  public Credentials vendAwsCredentials(CredentialContext context) {
+    S3StorageConfig s3StorageConfig = s3Configurations.get(context.getStorageBase());
```

Comment:
Does this require the `bucketPath` in the s3 config to be a bucket root path? Users may want to use a part of a bucket (instead of the whole bucket) in the storage config (which can be supported with a prefix match).

Reply:
Also, the paths stored in `s3Configurations` and in the context need to be normalized for the matching to be robust. This applies to URIs in Azure and GCP as well.

Reply:
Again, I didn't try and tackle much of anything of how the server properties, in particular the storage configurations, are stored in this PR. 

I think currently, letting people set paths beyond the bucket as the bucketPath makes it overly complicated with the way we are storing stuff in a properties file. Using a properties file and having to pattern match (what if it partially matches, etc.) is not great. I would wait to add this when 1) someone from the community requests it and 2) possibly when the storage configurations are not stored in a properties file.